artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/QueuePendingMessageMetrics.java

```diff
@@ -49,14 +52,18 @@
 
    private final Queue queue;
 
-   public QueuePendingMessageMetrics(final Queue queue) {
+   private final String name;
+
+   public QueuePendingMessageMetrics(final Queue queue, final String name) {
       Preconditions.checkNotNull(queue);
       this.queue = queue;
+      this.name = name;
    }
 
    public void incrementMetrics(final MessageReference reference) {
       long size = getPersistentSize(reference);
       COUNT_UPDATER.incrementAndGet(this);
+      logger.debugf("%s increment messageCount to %d: %s", this, messageCount, reference);
```

Comment:
can we put inside if debug enabled (as on hot path) so will be removed by jit if debug isnt on.

Reply:
Will fix.